### PR TITLE
Support for Dynamic Fields in Object

### LINKED ISSUE #1309: [Object] Support for Dynamic Fields in Object for Improved Composability
## Motivation

Currently, in Rooch, the underlying structure of `Table` is `Object<TableInfo>`, where `TableInfo` stores the root of smt. If we want to dynamically attach some data to the Object to facilitate easier composability, it would be necessary to include a `Table` or `TypeTable` field within `Object<T>`'s T. However, if Object supports dynamic fields, effectively carrying a Table by default and recording the root of the Table smt in ObjectEntity, implementing this feature becomes much more convenient.

## Proposal

1. Record the fields from `TableInfo` within `ObjectEntity`.
2. Merge the methods related to the `raw_table` module into the `object` module, making the `object` module the fundamental one.
3. Carry a default Global Object in `Context`, representing the Global state tree.
4. `Table` and `TypeTable` should be encapsulated based on the Object.
5. `Object<Account>` can replace the current `AccountStorage`.

## Reference

1. https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/sui-framework/sources/dynamic_field.move
2. https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/sui-framework/sources/dynamic_object_field.move
3. https://github.com/rooch-network/rooch/blob/main/frameworks/bitcoin-move/sources/utxo.move

### PR DESCRIPTION
**Pull request by baichuan3**
## Summary

Support for Dynamic Fields in Object.

1. Refactor Table implements, key type is no longer recorded, and records the fields from `TableInfo` within `ObjectEntity`.
2. Make raw_table the lowest foundation, and the Object's Dynamic field is based on the raw table. Making the object module the fundamental one.
3. Split `ObjectID` from `Object`, raw_table only relies on `ObjectID` module.
4. Added Object dynamic field module as Dynamic filed call entry.
5. Encapsulated `Table`, `ObjectTable`, and `TypeTable`  based on the Object.

TODO
1. Add object dynamic filed example
2. Replace the current `AccountStorage` with `Object<Account>`.

- Closes #1309

### CODE REVIEW

File: moveos/moveos-stdlib/moveos-stdlib/sources/object.move [hunk 1]
```diff
@@ -150,8 +112,10 @@ module moveos_std::object {
     /// This function is only can be called by the module of `T`.
     public fun remove<T: key>(self: Object<T>) : T {
         let Object{id} = self;
+        // TODO Need to ensure that the Table is empty before delete the Object?
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/object.move:115]: Yes

Comment (baichuan3) [reply moveos/moveos-stdlib/moveos-stdlib/sources/object.move:115]: `destroy_empty_table` in raw_table already check if the table is empty

**Review by jolestar — APPROVED** [review]

File: moveos/moveos-stdlib/moveos-stdlib/sources/raw_table.move [hunk 2]
```diff
@@ -129,31 +129,20 @@ module moveos_std::raw_table {
...
-        id: address,
-    }
-
-    public(friend) fun new_table_handle(id: address): TableHandle {
-        TableHandle { id }
-    }
-
     /// New a table. Aborts if the table exists.
-    native public(friend) fun new_table<K: copy + drop>(table_handle: TableHandle): TableInfo;
+    native public(friend) fun new_table(table_handle: ObjectID): TableInfo;
```
[the remark this answers is not in the document]
Comment (jolestar) [reply moveos/moveos-stdlib/moveos-stdlib/sources/raw_table.move:133]: The key is defined in Table<K,V>

[the remark this answers is not in the document]
Comment (baichuan3) [reply moveos/moveos-stdlib/moveos-stdlib/sources/raw_table.move:133]: By default, raw tables support any type of key and value.
Although <K, V> is specified when the upper Table is created, when adding elements to the table, the type of K is not checked, which means that arbitrary K and V are also received.

**Pull request merged**